File: ios/DepositoUrbano/Models/Space.swift
import Foundation
import CoreLocation

struct Space: Codable, Identifiable {
    let id: Int
    let hostId: Int
    let title: String
    let description: String
    let type: SpaceType
    let address: String
    let city: String
    let province: String
    let postalCode: String
    let latitude: Double
    let longitude: Double
    let size: Double
    let pricePerMonth: Double
    let pricePerDay: Double?
    let available: Bool
    let features: [String]
    let rules: String?
    let minBookingDays: Int
    let maxBookingDays: Int?
    let createdAt: Date?
    let updatedAt: Date?
    
    // Relations
    let host: User?
    let images: [SpaceImage]?
    let averageRating: Double?
    let reviewCount: Int?
    
    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
    }
    
    var mainImageUrl: String? {
        images?.first?.url
    }
}

enum SpaceType: String, Codable, CaseIterable {
    case room = "room"
    case garage = "garage"
    case warehouse = "warehouse"
    case locker = "locker"
    case other = "other"
    
    var displayName: String {
        switch self {
        case .room:
            return "Habitación"
        case .garage:
            return "Garage"
        case .warehouse:
            return "Depósito"
        case .locker:
            return "Baulera"
        case .other:
            return "Otro"
        }
    }
    
    var icon: String {
        switch self {
        case .room:
            return "bed.double.fill"
        case .garage:
            return "car.fill"
        case .warehouse:
            return "shippingbox.fill"
        case .locker:
            return "lock.fill"
        case .other:
            return "questionmark.square.fill"
        }
    }
}

struct SpaceImage: Codable, Identifiable {
    let id: Int
    let spaceId: Int
    let url: String
    let order: Int
}

File: backend/src/models/Space.ts
import { DataTypes, Model, Sequelize } from 'sequelize';

export interface SpaceAttributes {
  id: number;
  hostId: number;
  title: string;
  description: string;
  type: 'room' | 'garage' | 'warehouse' | 'locker' | 'other';
  address: string;
  city: string;
  province: string;
  postalCode: string;
  latitude: number;
  longitude: number;
  size: number; // in square meters
  pricePerMonth: number;
  pricePerDay?: number;
  available: boolean;
  features: string[]; // JSON array of features
  rules?: string;
  minBookingDays: number;
  maxBookingDays?: number;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface SpaceInstance extends Model<SpaceAttributes>, SpaceAttributes {}

export const SpaceModel = (sequelize: Sequelize) => {
  const Space = sequelize.define<SpaceInstance>('Space', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    hostId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'users',
        key: 'id'
      }
    },
    title: {
      type: DataTypes.STRING,
      allowNull: false
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    type: {
      type: DataTypes.ENUM('room', 'garage', 'warehouse', 'locker', 'other'),
      allowNull: false
    },
    address: {
      type: DataTypes.STRING,
      allowNull: false
    },
    city: {
      type: DataTypes.STRING,
      allowNull: false
    },
    province: {
      type: DataTypes.STRING,
      allowNull: false
    },
    postalCode: {
      type: DataTypes.STRING,
      allowNull: false
    },
    latitude: {
      type: DataTypes.DECIMAL(10, 8),
      allowNull: false
    },
    longitude: {
      type: DataTypes.DECIMAL(11, 8),
      allowNull: false
    },
    size: {
      type: DataTypes.FLOAT,
      allowNull: false,
      validate: {
        min: 0.1
      }
    },
    pricePerMonth: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
      validate: {
        min: 0
      }
    },
    pricePerDay: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true,
      validate: {
        min: 0
      }
    },
    available: {
      type: DataTypes.BOOLEAN,
      defaultValue: true
    },
    features: {
      type: DataTypes.JSON,
      defaultValue: []
    },
    rules: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    minBookingDays: {
      type: DataTypes.INTEGER,
      defaultValue: 1,
      validate: {
        min: 1
      }
    },
    maxBookingDays: {
      type: DataTypes.INTEGER,
      allowNull: true,
      validate: {
        min: 1
      }
    }
  }, {
    tableName: 'spaces',
    timestamps: true,
    underscored: true
  });

  return Space;
};